x-pack/plugins/actions/server/lib/get_execution_log_aggregation.ts

```diff
@@ -0,0 +1,446 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KueryNode } from '@kbn/es-query';
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import Boom from '@hapi/boom';
+import { flatMap, get, isEmpty } from 'lodash';
+import { AggregateEventsBySavedObjectResult } from '@kbn/event-log-plugin/server';
+import { fromKueryExpression, toElasticsearchQuery } from '@kbn/es-query';
+import { IExecutionLog, IExecutionLogResult, EMPTY_EXECUTION_KPI_RESULT } from '../../common';
+
+const DEFAULT_MAX_BUCKETS_LIMIT = 1000; // do not retrieve more than this number of executions
+const DEFAULT_MAX_KPI_BUCKETS_LIMIT = 10000;
+
+const SPACE_ID_FIELD = 'kibana.space_ids';
+const ACTION_NAME_FIELD = 'kibana.action.name';
+const START_FIELD = 'event.start';
+const ACTION_FIELD = 'event.action';
+const OUTCOME_FIELD = 'event.outcome';
+const DURATION_FIELD = 'event.duration';
+const MESSAGE_FIELD = 'message';
+const VERSION_FIELD = 'kibana.version';
+const ERROR_MESSAGE_FIELD = 'error.message';
+const SCHEDULE_DELAY_FIELD = 'kibana.task.schedule_delay';
+const EXECUTION_UUID_FIELD = 'kibana.action.execution.uuid';
+
+const Millis2Nanos = 1000 * 1000;
+
+export const ACTION_FILTER = 'event.provider: actions AND NOT event.action: execute-start';
+export const EMPTY_EXECUTION_LOG_RESULT = {
+  total: 0,
+  data: [],
+};
+
+interface IActionExecution
+  extends estypes.AggregationsTermsAggregateBase<{ key: string; doc_count: number }> {
+  buckets: Array<{ key: string; doc_count: number }>;
+}
+
+interface IExecutionUuidKpiAggBucket extends estypes.AggregationsStringTermsBucketKeys {
```

Comment:
leftover from the rule execution code?

Reply:
Resolved in this commit 5104f14d5b742547cfeff3c233988eb90ae1f484